lib-gphotos/client.go

```diff
@@ -136,26 +203,26 @@ func (client *Client) UploadFile(filePath string, pAlbumID ...string) (*photosli
 				retryCount++
 				continue
 			}
-			return nil, stacktrace.Propagate(err, "failed adding media %s", filename)
+			return nil, uploadURL, stacktrace.Propagate(err, "failed adding media %s", filePath)
 		}
 
 		if batchResponse == nil || len(batchResponse.NewMediaItemResults) != 1 {
-			return nil, stacktrace.NewError("len(batchResults) should be 1")
+			return nil, uploadURL, stacktrace.NewError("len(batchResults) should be 1")
 		}
 		result := batchResponse.NewMediaItemResults[0]
 		if result.Status.Message != "OK" && result.Status.Message != "Success" {
 			// TODO: We should use a different field like `googleapi.ServerResponse`
-			return nil, stacktrace.NewError("status message should be OK/Succecc, found: %s", result.Status.Message)
+			return nil, uploadURL, stacktrace.NewError("status message should be OK/Succecc, found: %s", result.Status.Message)
 		}
 
-		log.Printf("%s uploaded successfully as %s", filename, result.MediaItem.Id)
-		return result.MediaItem, nil
+		log.Printf("%s uploaded successfully as %s", filePath, result.MediaItem.Id)
+		return result.MediaItem, "", nil
 	}
-	return nil, nil
+	return nil, uploadURL, nil
```

Comment:
It's strange that you put an `uploadURL` when you call the function and the same function returns another `uploadURL`... WDYT?

Reply:
`uploadURL` can be empty when the function is first called. Splitting the function will certainly help make that clearer.

Reply:
Also, the [passed `uploadURL` can be reset](https://github.com/gphotosuploader/google-photos-api-client-go/pull/5/files/80525579be58e07d3494ac4c20b3bb916c427151#diff-a489c843c86bd57cb8eab12f172d78f1R90) if the corresponding status is not `active`

Reply:
I've reworked that part to pass `uploadURL` as string pointer that is updated.